spec/controllers/admin/subscriptions_controller_spec.rb

```diff
@@ -458,7 +458,11 @@
             }
             let!(:order) { proxy_order.initialise_order! }
 
-            before { break unless order.next! while !order.completed? }
+            before do
+              while !order.completed?
+                break unless order.next!
+              end
+            end
```

Comment:
Wow, I didn't realise this code was repeated so much in the specs. I think it would be great to move it to a helper method.

In fact, we already have one in the app: `Orders::WorkflowService#complete`. This has a little more logic, so it could potentially affect some specs. But I think it would be worth trying, because it would leave the data in a more realistic state, making for more realistic tests.
Do any other @openfoodfoundation/reviewers agree?

Reply:
I agree, let's re-use code where possible and DRY this snippet. At least a spec helper would be good.

Reply:
Thanks @dacook, I went with your suggestion and updated the specs to use `Orders::WorkflowService#complete!`.

I chose the bang version since it calls `order.next!` under the hood, which matches what the specs were doing before. I also used `advance_to_payment` where appropriate.

There's no "`advance_to_delivery`", and since that step is only used in a couple of places, I kept the while loop with a small cleanup for readability solving the rubocop offense.